var/spack/repos/builtin/packages/cxx/package.py

```diff
@@ -34,8 +35,13 @@ def test(self):
             cxx_opts = [compiler.cxx11_flag] if "c++11" in test else []
 
             cxx_opts += ["-o", exe_name, filepath]
-            compiled = self.run_test(cxx_exe, options=cxx_opts, installed=True)
+            cxx_exe = which(join_path(self.prefix.bin, cxx_exe))
```

Comment:
I believe this and line 25 should be moved above the for-loop at line 21 since the environment variable isn't going to change during the iteration.